### PR TITLE
Update List.IndexOf node description

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Per customer request,  updating node description for `List.IndexOf` node. See UI change below (notice word `hard` was later refactored to be `strict`):

image

image

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@DynamoDS/dynamo  @Amoursol

### CODE REVIEW

File: src/Libraries/CoreNodes/List.cs [hunk 1]
```diff
@@ -163,11 +163,11 @@ public static IList SetUnion(IList<object> list1, IList<object> list2)
         }
 
         /// <summary>
-        ///     Returns the index of the element in the given list.
+        ///     Returns the index of the element in the given list. Match between given list and target element must be a hard match (i.e. int to int, double to double, string to string, object to object etc.)
         /// </summary>
         /// <param name="list">The list to find the element in.</param>
         /// <param name="element">The element whose index is to be returned.</param>
-        /// <returns name="int">The index of the element in the list.</returns>
+        /// <returns name="int">The index of the element in the list. Invalid index -1 will be returned if hard match not found.</returns>
```
Comment (aparajit-pratap) [inline src/Libraries/CoreNodes/List.cs:170]: how about saying "strict match"?

Comment (Amoursol) [reply src/Libraries/CoreNodes/List.cs:170]: Agree that strict is a better term!

File: src/Libraries/CoreNodes/List.cs [hunk 2]
```diff
@@ -163,11 +163,11 @@ public static IList SetUnion(IList<object> list1, IList<object> list2)
         }
 
         /// <summary>
-        ///     Returns the index of the element in the given list.
+        ///     Returns the index of the element in the given list. Match between given list and target element must be a hard match (i.e. int to int, double to double, string to string, object to object etc.)
```
Comment (Amoursol) [inline src/Libraries/CoreNodes/List.cs:166]: Ditto here for: Hard --> Strict

**Comment by QilongTang** [comment]
@Amoursol @aparajit-pratap Addressed the comments. If looks good, please approve.

**Review by zeusongit — APPROVED** [review]

**Comment by Amoursol** [comment]
LGTM @QilongTang - did you need me to officially commit my review?

**Pull request merged**